addons/mail/tests/discuss/test_discuss_attachment_controller.py

```diff
@@ -23,3 +23,23 @@ def test_attachment_allowed_upload_public_channel(self):
                 (self.user_public, True),
             ),
         )
+
+    def test_attachment_delete_linked_to_channel(self):
+        """Test access to delete an attachment associated with a channel"""
+        channel = self.env["discuss.channel"].create({"name": "public channel"})
+        # Subtest format: (user, token, result)
+        self._execute_subtests_delete(
+            (
+                (self.guest, False, False),
+                (self.guest, True, False),
+                (self.user_admin, False, True),
+                (self.user_admin, True, True),
+                (self.user_employee, False, True),
+                (self.user_employee, True, True),
+                (self.user_portal, False, False),
+                (self.user_portal, True, False),
+                (self.user_public, False, False),
+                (self.user_public, True, False),
```

Comment:
As discussed we might significantly reduce the size of these tests without reducing coverage by using `itertools.product` to group similar results together.